apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts

```diff
@@ -59,71 +60,111 @@ const addCourseToSharingSetPopover = (resLocals, sharing_set) => {
-                <div class="modal-body">
-                  <p class="form-text">
-                    <strong>Unable to change your course's sharing name.</strong>
-                  </p>
-                  <p>
-                    Your course's sharing name cannot be changed because at least one question has
-                    been shared. Doing so would break the assessments of other courses that have
-                    imported your questions.
-                  </p>
-                </div>
-              `}
-        </div>
+interface SharingSetModalProps {
+  csrfToken: string;
+  title: string;
+  sharing_set: { name: string; id: string; deletable: boolean };
+}
+
+/*
+ * TEST, better way to choose body and footer based on deletable?
+ * Doing it with the " ? : " operator makes the Modal appear as an arrow tag above the button.
+*/
+function deleteSharingSetModal(sharing_set, csrfToken) {
+  let body = '';
+  let footer = '';
+  if (sharing_set.deletable) {
+    body = html`
+      <p><strong>This action cannot be undone.</strong></p>
+      <p>
+        Are you sure you would like to delete the sharing set "${sharing_set.name}"?
+      </p>
+    `;
+    footer = html`
+      <input type="hidden" name="__action" value="delete_sharing_set" />
+      <input type="hidden" name="sharing_set_id" value="${sharing_set.id}" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+      <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+      <button type="submit" class="btn btn-primary">Delete Sharing Set</button>
+    `;
+  } else {
+    body = html`
+      <p><strong>Unable to delete sharing set.</strong></p>
+      <p>
+        This sharing set cannot be deleted because it has been shared 
+        and at least one question has been added. Doing so would break the 
+        assessments of other courses that have imported your questions.
+      </p>
+      `;
+    footer = html`
+      <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+    `;
+  }
+  return Modal({
+    title: 'Delete Sharing Set',
+    id: `deleteSharingSetModal-${sharing_set.id}`,
+    body: body,
+    footer: footer,
+    size: 'default',
+  });
+}
+
+/*
+ * TEST, better way to choose body and footer based on deletable?
+ * Doing it with the " ? : " operator makes the Modal appear as an arrow tag above the button.
+*/
+function chooseSharingNameModal(canChooseSharingName, csrfToken) {
+  let body = '';
+  let footer = '';
+  if (canChooseSharingName) {
+    body = html`
+      <p class="form-text">Enter the sharing name you would like for your course.</p>
+      <div>
+        <label for="course_sharing_name">Enter Sharing Name</label>
+        <input class="form-control" type="text" name="course_sharing_name" required />
       </div>
-    </div>
-  `;
-};
+      <p>
+        <strong>
+          Once you have shared a question either publicly or with another course, you
```

Comment:
Hi @Michael-Holmes-SDE, thanks for refactoring this to use the Modal compenent. Can you put this refactor of the choose sharing name modal into a separate branch/PR than the code to allow deleting sharing sets? These changes aren't really related to one another so they shouldn't be in the same PR. Thanks!